source/simulator/nullspace.cc

```diff
@@ -165,6 +165,54 @@ namespace aspect
 
   }
 
+  namespace
+  {
+    inline MPI_Datatype mpi_type_id (const unsigned int *)
+    {
+      return MPI_UNSIGNED;
+    }
+
+
+    inline MPI_Datatype mpi_type_id (const unsigned long int *)
+    {
+      return MPI_UNSIGNED_LONG;
+    }
+  }
+
+  template <int dim>
+  void Simulator<dim>::setup_nullspace_constraints(ConstraintMatrix &constraints)
+  {
+    const unsigned int flags[] = {(NullspaceRemoval::linear_momentum_x
+                                   |NullspaceRemoval::net_translation_x),
+                                  (NullspaceRemoval::linear_momentum_y
+                                   |NullspaceRemoval::net_translation_y),
+                                  (NullspaceRemoval::linear_momentum_z
+                                   |NullspaceRemoval::net_translation_z)
+                                 };
+
+    for (unsigned int d=0; d<dim; ++d)
+      if (parameters.nullspace_removal & flags[d])
+        {
+          // find the first candidate for a constraint in direction d that is
+          // not constrained yet
+          types::global_dof_index idx = d;
+          while (!constraints.can_store_line(idx) || constraints.is_constrained(idx))
+            {
+              idx += dim;
+            }
+
+          // Make a reduction to find the smallest index (processors with a larger candidate
+          // just happened to not be able to store that index)
+          types::global_dof_index global_idx;
+          MPI_Allreduce(&idx, &global_idx, 1, mpi_type_id(&global_idx), MPI_MIN, mpi_communicator);
```

Comment:
`Utilities::MPI::min` should do, no? (Ah, bummer -- I see that we only have `max` but not `min` :-( )